File: 평강/app/project/src/main/java/com/eventpurchase/project/global/common/exception/GlobalExceptionHandler.java
package com.eventpurchase.project.global.common.exception;

import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import com.eventpurchase.project.global.common.response.ErrorResponse;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@Slf4j
@RestControllerAdvice
public class GlobalExceptionHandler {

    @ExceptionHandler(CustomException.class)
    public ResponseEntity<ErrorResponse>  handleCustomException(CustomException ex){
        log.error("CustomException 발생: {}", ex.getErrorCode().getMessage());
        return ResponseEntity.status(ex.getErrorCode().getStatus()).body(ErrorResponse.of(ex.getErrorCode()));
    }
}

File: 평강/app/project/application-api/src/main/kotlin/com/eventpurchase/project/GlobalExceptionHandler.kt
package com.eventpurchase.project

import com.eventpurchase.project.cart.CartItemNotFoundException
import com.eventpurchase.project.cart.ProductNotAvailableException
import com.eventpurchase.project.product.ProductNotFoundException
import com.eventpurchase.project.shared.response.ErrorResponse
import com.eventpurchase.project.user.UserNotFoundException
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(ProductNotFoundException::class)
    fun handleProductNotFound(e: ProductNotFoundException): ResponseEntity<ErrorResponse> =
        ResponseEntity.status(404).body(ErrorResponse(404, "상품을 찾을 수 없습니다."))

    @ExceptionHandler(ProductNotAvailableException::class)
    fun handleProductNotAvailable(e: ProductNotAvailableException): ResponseEntity<ErrorResponse> =
        ResponseEntity.status(404).body(ErrorResponse(404, "상품을 찾을 수 없습니다."))

    @ExceptionHandler(UserNotFoundException::class)
    fun handleUserNotFound(e: UserNotFoundException): ResponseEntity<ErrorResponse> =
        ResponseEntity.status(404).body(ErrorResponse(404, "유저를 찾을 수 없습니다."))

    @ExceptionHandler(CartItemNotFoundException::class)
    fun handleCartItemNotFound(e: CartItemNotFoundException): ResponseEntity<ErrorResponse> =
        ResponseEntity.status(404).body(ErrorResponse(404, "장바구니 상품을 찾을 수 없습니다."))

    @ExceptionHandler(IllegalArgumentException::class)
    fun handleIllegalArgument(e: IllegalArgumentException): ResponseEntity<ErrorResponse> =
        ResponseEntity.status(400).body(ErrorResponse(400, e.message ?: "잘못된 요청입니다."))

    @ExceptionHandler(Exception::class)
    fun handleException(e: Exception): ResponseEntity<ErrorResponse> =
        ResponseEntity.status(500).body(ErrorResponse(500, "서버 오류가 발생했습니다."))
}
